models/demos/mamba/tt/mamba_one_step_ssm.py

```diff
@@ -74,17 +74,14 @@ def preprocess_C(x):
         self.dt_proj_weights = load_fn(dt_proj_weight_name, lambda x: x.transpose(-1, -2), tt_dtype=ttnn.bfloat8_b)
         self.dt_proj_bias = load_fn(dt_proj_bias_name, tt_dtype=ttnn.bfloat8_b)
 
-        # B_intermediate_tranform_weights = torch.eye(self.n).repeat(1, self.hidden_size).unsqueeze(0).unsqueeze(0)
-
-        # A weight
         A_weight_name = "mixer.A_log"
 
         def preprocess_A(x):
             x = -torch.exp(x.float())
             # padding with inf
-            # x = F.pad(x, (0, 16), "constant", float("-inf"))
-            x = x.reshape(1, self.hidden_size * self.n)  # (1, 2en)
-            return x.repeat(self.batch_size, 1)  # b, 2en
+            x = torch.nn.functional.pad(x, (0, 16), "constant", float("-inf"))
+            # x = x.reshape(1, self.hidden_size * self.n)  # (1, 2en)
```

Comment:
Do we need to depend on torch within the tt version of Mamba?

Reply:
Unfortunately yes, the slicing and padding we require to preprocess our weights is not provided by `ttnn`

Reply:
Does ttnn.pad not work for you?

Reply:
Can you give me a pytest for this showing it fails?  If so, I'd like to get this fixed asap.

Reply:
This code implements the padding part of the preprocessing:
```python
import torch
import ttnn


def test_preprocess_weight(device: ttnn.Device):
    batch = 32
    latent = 16
    hidden_size = 5120
    x = torch.rand((1, 1, hidden_size, latent)) # from state_dict
    x = ttnn.from_torch(
        x, dtype=ttnn.bfloat16, device=device, layout=ttnn.TILE_LAYOUT, memory_config=ttnn.DRAM_MEMORY_CONFIG
    )
    y = ttnn.pad(x, ((0, 0), (0, 0), (0, 0), (0, 16)), float("-inf"))
    # y = y[:, :, 0, :].repeat(1, 1, batch, 1)
```

I don't know if the slice (`y = y[:, :, 0, :]`) is supported by ttnn. 

This preprocessing is done at initialization time on the host when we pull the weights from the HF state dictionary.

Reply:
We fallback to torch at the moment to do that sort of slicing.   The intent was to have everying use this slicing and then we will circle back and fix it at some point.